account_payment_term_cutoff_day/models/account_payment_term_line.py

```diff
@@ -23,5 +24,7 @@ def _get_due_date(self, date_ref):
         if date_ref and self.end_month and self.cutoff_day:
             date_dt = fields.Date.to_date(date_ref)
             if date_dt.day > self.cutoff_day:
-                due_date += relativedelta(months=1)
+                due_date = date_utils.end_of(
+                    due_date + relativedelta(months=1), "month"
+                )
```

Comment:
Can we do as follows instead, as it should better reflect the concept of the cutoff day?
```.py
    def _get_due_date(self, date_ref):
        if date_ref and self.cutoff_day:
            date_dt = fields.Date.to_date(date_ref)
            if date_dt.day > self.cutoff_day:
                date_ref = date_ref + relativedelta(months=1)
        return super()._get_due_date(date_ref)
```

Reply:
It is valid, and I also added `self.end_month` in the condition to block the intended behavior if the user enables `end_month` and sets up a `cutoff_day`, then later disables `end_month` for some reason.